pandas/core/reshape/melt.py

```diff
@@ -217,7 +218,15 @@ def melt(
                 frame.columns.name if frame.columns.name is not None else "variable"
             ]
     elif is_list_like(var_name):
-        raise ValueError(f"{var_name=} must be a scalar.")
+        if isinstance(frame.columns, MultiIndex):
+            var_name = list(var_name)
```

Comment:
Is the `list` cast needed here? it should be `Sized` to compute `len`

Reply:
Ah true. Could you use `is_iterator` to check for the single use object and cast to `list` if so?